### PR TITLE
Trigger videoTimingInfo on transmuxer with relevant timing info and prepended GOP duration

### PR DESCRIPTION
**Pull request by gesinger**
The video timing info can be useful for determining the duration of a segment (both presentation and decode), as well as any time prepended due to the addition of GOPs for alignment.

### CODE REVIEW

File: lib/mp4/transmuxer.js [hunk 1]
```diff
@@ -923,6 +969,8 @@ Transmuxer = function(options) {
 
           pipeline.videoSegmentStream.on('processedGopsInfo',
             self.trigger.bind(self, 'gopInfo'));
+          pipeline.videoSegmentStream.on('segmentTimingInfo',
+            self.trigger.bind(self, 'videoSegmentTimingInfo'));
```
[the remark this answers is not in the document]
Comment (gesinger) [reply lib/mp4/transmuxer.js]: Yeah, we'll change them in TB4A. Unless you prefer videoTimingInfo over videoSegmentTimingInfo

File: lib/mp4/transmuxer.js [hunk 2]
```diff
@@ -62,6 +62,34 @@ var arrayEquals = function(a, b) {
...
+    end: {
+      dts: endDts,
+      pts: endPts
+    }
+  };
+
+  if (prependedGops) {
+    // Use presentation timestamp as calculations are based on showing frames, rather than
+    // the underlying decoding frames.
+    segmentTimingInfo.prependedContentDuration = originalStartPts - prependedGops[0].pts;
```
[the remark this answers is not in the document]
Comment (gesinger) [reply lib/mp4/transmuxer.js]: Just in terms of defensive programming, or somewhere specific that those values can be passed in?

[the remark this answers is not in the document]
Comment (ldayananda) [reply lib/mp4/transmuxer.js]: I think right now it's possible that we'll pass in an empty array for `prependedGops`

File: lib/mp4/transmuxer.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (ldayananda) [reply lib/mp4/transmuxer.js]: changing them in TB4A works!

File: lib/mp4/transmuxer.js [hunk 3]
```diff
@@ -67,27 +67,19 @@ var generateVideoSegmentTimingInfo = function(
...
     start: {
       dts: startDts,
       pts: startPts
     },
     end: {
       dts: endDts,
       pts: endPts
-    }
+    },
+    prependedContentDuration: prependedContentDuration
```
[the remark this answers is not in the document]
Comment (ldayananda) [reply lib/mp4/transmuxer.js]: actually, object shorthand isn't allowed in this project yet.

**Pull request merged**